solutions/components-in-depth/src/app/counter-display/counter-display.component.ts

```diff
@@ -0,0 +1,22 @@
+import { Component } from '@angular/core';
+
+const defaultCounterValue = 10;
+
+@Component({
+  selector: 'app-counter-display',
+  templateUrl: 'counter-display.component.html'
+})
+export class CounterDisplayComponent {
+  counterValue: number;
+  defaultCounterValue: number;
+
+  ngOnInit() {
+    this.setDefaultValue();
+  }
+
+  setDefaultValue() {
+    this.counterValue = this.defaultCounterValue =
+      Number.isInteger(this.defaultCounterValue) ?
+      this.defaultCounterValue : defaultCounterValue;
```

Comment:
For the purposes of the solution, this line is a little too compact, and not much is gained by it.
Can we be more declarative here?

```
this.defaultCounterValue = Number.isInteger(this.defaultCounterValue) ?
        this.defaultCounterValue : defaultCounterValue;
this.counterValue = this.defaultCounterValue;
```